drivers/spi/spi_silabs_eusart.c

```diff
@@ -1,11 +1,16 @@
 /*
  * Copyright (c) 2024 Daikin Comfort Technologies North America, Inc.
+ * Copyright (c) 2025 Silicon Laboratories Inc.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #define DT_DRV_COMPAT silabs_eusart_spi
 
+#if defined(CONFIG_SPI_ASYNC)
+#define SPI_SILABS_SPI_EUSART_DMA (1U)
+#endif
```

Comment:
It looks to me like you're using this new define as a kind of feature switch, even though it's not a Kconfig option. I'd suggest to make it a proper Kconfig option instead. It can be promptless and auto-enabled whenever `CONFIG_SPI_ASYNC` is enabled, to keep the same semantics as here, i.e. something like the following in `Kconfig.silabs_eusart`:
```
config SPI_SILABS_EUSART_DMA
	depends on SPI_SILABS_EUSART
	default y if SPI_ASYNC
```

Reply:
This description doesn't entirely match my suggestion, DMA should be usable independently of SPI_ASYNC=y.
What I tried to suggest is as follows: It is possible to use the presence of a `dmas` property in DTS to decide whether a _specific_ instance should use DMA or not. When it comes to globally enabling DMA _support_ in the driver, this can be done either through a Kconfig option as @jhedberg suggests or by looking at some DTS property (such as checking if _any_ SPI node has a `dmas` property).

Looking at drivers in the tree, it seems like _most_ drivers do this with Kconfig, while some drivers compile it in unconditionally and only use DTS to decide whether a specific driver instance should use it or not. My conclusion was therefore to go the Kconfig route.